src/contract/tokens.rs

```diff
@@ -298,9 +298,11 @@ tokenizable_item! {
 impl Tokenizable for Vec<u8> {
     fn from_token(token: Token) -> Result<Self, Error> {
         match token {
-            Token::Bytes(data) => Ok(data),
-            Token::FixedBytes(data) => Ok(data),
-            other => Err(Error::InvalidOutputType(format!("Expected `bytes`, got {:?}", other))),
+            Token::Bytes(data) | Token::FixedBytes(data) => Ok(data),
+            Token::Array(tokens) | Token::FixedArray(tokens) => {
```

Comment:
The problem with this is that you will decode `Vec<u8>` from `Token::Array`, but encoding (`into_tokens`) will always return `Token::Bytes` which is quite misleading.

The split is done on purpose currently (also a bit due to Rust limitations). I think the best way forward would be to introduce a type that looks like this:
```rust
pub struct BytesArray(pub Vec<u8>);
impl Tokenizable for BytesArray {
    fn from_token(token: Token) -> Result<Self, Error> {
        match token {
            Token::FixedArray(tokens) | Token::Array(tokens) => {
                Ok(Self(tokens.into_iter().map(Tokenizable::from_token).collect()?))
            }
            other => Err(Error::InvalidOutputType(format!("Expected `Array`, got {:?}", other))),
        }
    }

    fn into_token(self) -> Token {
        Token::Array(self.into_iter().map(Tokenizable::into_token).collect())
    }
}
```

Unless `Token::Bytes` and `Token::Array` have the same ethabi encoding and there is no confusion here (I really don't remember).

Also please add tests.